server/tserver/src/main/java/org/apache/accumulo/tserver/compaction/EverythingCompactionStrategy.java

```diff
@@ -27,7 +27,7 @@
 
   @Override
   public boolean shouldCompact(MajorCompactionRequest request) throws IOException {
-    return request.getFiles().size() > 0;
+      return true; // ACCUMULO-3645 compact for empty files too
```

Comment:
The only time this will make a difference is when the USER manually issues a compaction with at least one iterator, in which case the user should expect the compaction occurs by default.

I will add a condition to the Tablet code to keep the "optimized" behavior for empty tablets on user-issued compactions that do not have any iterators configured.